src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -462,7 +462,7 @@ class MySimulation : public SimulationHelper<simDim>
         auto fieldE = dc.get< FieldE >( FieldE::getName(), true );
         auto fieldB = dc.get< FieldB >( FieldB::getName(), true );
 
-        if( step != 0 )
+        if( this->restartRequested )
         {
             namespace nvfct = PMacc::nvidia::functors;
 
```

Comment:
if the intention was to keep the logic identical, I think be two background field below need to be called with `step + 1u` now